helix-term/src/commands.rs

```diff
@@ -435,7 +435,11 @@ pub fn replace(cx: &mut Context) {
 
             let transaction =
                 Transaction::change_by_selection(doc.text(), doc.selection(view.id), |range| {
-                    (range.from(), range.to() + 1, Some(text.clone()))
+                    (
+                        range.from(),
+                        range.to() + 1,
```

Comment:
That was quick! Since this function doesn't have proper bounds checks, can you add that? Should be identical to https://github.com/helix-editor/helix/blob/0201ef920516db0fd4a4e6939f87bd0d7e4cf5dd/helix-term/src/commands.rs#L794-L796

Reply:
Done :)